__tests__/app.test.js

```diff
@@ -123,7 +123,46 @@ describe("GET /api/articles", () => {
         });
       });
   });
+  test("GET 200 responds with an array of the articles with the requested topic.", () => {
+    return request(app)
+      .get("/api/articles?topic=cats")
+      .expect(200)
+      .then(({ body }) => {
+        expect(body.articles).toHaveLength(1);
+        body.articles.forEach((article) => {
+          expect(article).toMatchObject({
+            article_id: expect.any(Number),
+            title: expect.any(String),
+            topic: expect.any(String),
+            author: expect.any(String),
+            created_at: expect.any(String),
+            votes: expect.any(Number),
+            article_img_url: expect.any(String),
+            comment_count: expect.any(String),
+          });
+        });
+      });
+  });
+  test("GET 200 responds with an array of the articles with the requested topic.", () => {
+    return request(app)
+      .get("/api/articles?topic=mitch")
+      .expect(200)
+      .then(({ body }) => {
+        expect(body.articles).toHaveLength(12);
+      });
+  });
+  test("GET 404 responds with an error message when there is no article available under this topic", () => {
+    return request(app)
+      .get("/api/articles?topic=paper")
+      .expect(404)
+      .then((response) => {
+        expect(response.body.msg).toEqual(
+          "There is currently no article with this topic available"
+        );
+      });
+  });
```

Comment:
Nice one, we may also want to consider a valid topic that exists but happens to have no articles associated with it.

Reply:
@nc-richard-neat Thanks, for the feedback. This one tests for an existing topic that has no articles available.  Am I missing a different test?